purchase_order_line_schedule/models/purchase_order_line_schedule.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2021 ForgeFlow, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models
+
+
+class PurchaseOrderLineSchedule(models.Model):
+
+    _name = "purchase.order.line.schedule"
+    _description = "Purchase Order Line Schedule"
+    _order = "order_line_id, date_planned, id"
+
+    order_id = fields.Many2one(
+        comodel_name="purchase.order", related="order_line_id.order_id", store=True
+    )
+    purchase_state = fields.Selection(related="order_line_id.order_id.state",)
+    qty_received_method = fields.Selection(related="order_line_id.qty_received_method",)
+    order_line_id = fields.Many2one(
+        comodel_name="purchase.order.line", required=True, ondelete="cascade"
+    )
+    date_planned = fields.Datetime(string="Scheduled Date", index=True, required=True)
+    product_qty = fields.Float(
+        string="Quantity", digits="Product Unit of Measure", required=True
+    )
+    qty_received = fields.Float(
+        "Received",
+        compute="_compute_qty_received",
+        inverse="_inverse_qty_received",
+        compute_sudo=True,
+        digits="Product Unit of Measure",
+    )
+    qty_to_receive = fields.Float(
+        "To Receive",
+        compute="_compute_qty_to_receive",
+        compute_sudo=True,
+        digits="Product Unit of Measure",
+    )
```

Comment:
Should negative quantities be allowed? Or is in addition, i.e. purchase_order_line_schedule_rental

Reply:
This is necessary to ensure compatibility with existing modules / flows. If a user changes the quantity to order by default it allocates the extra quantity to the most recent schedule line. The user can always to go to the schedule lines and adjust.